### PR TITLE
feat: Transfer support custom dropdown Icon

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

- #43770

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | feat: Transfer support custom dropdown Icon |
| 🇨🇳 Chinese | Transfer 组件支持自定义 dropdown 下拉图标 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 8964f93</samp>

This pull request adds a new feature to the `Transfer` component, which lets users customize the icon of the dropdown menu in the transfer list. It modifies the `TransferList` component, the TypeScript definition file, and the API documentation in both English and Chinese.

### <samp>🤖 Generated by Copilot at 8964f93</samp>

*  Add `dropdownIcon` property to `Transfer` and `TransferList` components to allow customizing the dropdown icon of the transfer list header ([link](https://github.com/ant-design/ant-design/pull/43773/files?diff=unified&w=0#diff-c165dbeb24339c042fcc975a9b661af8947ea942457cf8935f29817ac86033a9R43), [link](https://github.com/ant-design/ant-design/pull/43773/files?diff=unified&w=0#diff-16f601b16a2f3b1d448af36a6b569c1d47bf9e2c351318a3b986c42fcbc5b7fdR108), [link](https://github.com/ant-design/ant-design/pull/43773/files?diff=unified&w=0#diff-16f601b16a2f3b1d448af36a6b569c1d47bf9e2c351318a3b986c42fcbc5b7fdR134), [link](https://github.com/ant-design/ant-design/pull/43773/files?diff=unified&w=0#diff-16f601b16a2f3b1d448af36a6b569c1d47bf9e2c351318a3b986c42fcbc5b7fdR379), [link](https://github.com/ant-design/ant-design/pull/43773/files?diff=unified&w=0#diff-16f601b16a2f3b1d448af36a6b569c1d47bf9e2c351318a3b986c42fcbc5b7fdR418), [link](https://github.com/ant-design/ant-design/pull/43773/files?diff=unified&w=0#diff-f25eb82686b163c8fb19e3d453b3c168ce5311046d76e7b3a1192173b676ea97R46), [link](https://github.com/ant-design/ant-design/pull/43773/files?diff=unified&w=0#diff-c6feaf8f7f09d7e3901905a89bc11ecd461a7340d5a8e8ea4dbc994a77bdc415R77), [link](https://github.com/ant-design/ant-design/pull/43773/files?diff=unified&w=0#diff-c6feaf8f7f09d7e3901905a89bc11ecd461a7340d5a8e8ea4dbc994a77bdc415R105), [link](https://github.com/ant-design/ant-design/pull/43773/files?diff=unified&w=0#diff-c6feaf8f7f09d7e3901905a89bc11ecd461a7340d5a8e8ea4dbc994a77bdc415L355-R359))
*  Define `isValidIcon` function in `list.tsx` to check if a given icon is valid for rendering ([link](https://github.com/ant-design/ant-design/pull/43773/files?diff=unified&w=0#diff-c6feaf8f7f09d7e3901905a89bc11ecd461a7340d5a8e8ea4dbc994a77bdc415R37-R38))

### CODE REVIEW

File: components/transfer/list.tsx [hunk 1]
```diff
@@ -34,6 +34,8 @@ function getEnabledItemKeys<RecordType extends KeyWiseTransferItem>(items: Recor
   return items.filter((data) => !data.disabled).map((data) => data.key);
 }
 
+const isValidIcon = (icon: React.ReactNode) => icon !== undefined && icon !== false;
```
Comment (MadCcc) [inline components/transfer/list.tsx:37]: 检查 undefined 就够了

File: components/transfer/index.tsx [hunk 2]
```diff
@@ -105,6 +105,7 @@ export interface TransferProps<RecordType> {
   oneWay?: boolean;
   pagination?: PaginationType;
   status?: InputStatus;
+  dropdownIcon?: React.ReactNode;
```
Comment (MadCcc) [inline components/transfer/index.tsx:108]: dropdownIcon 语义太宽泛了，需要再特定一下

Comment (MadCcc) [reply components/transfer/index.tsx:108]: selectionIcon
和 Table 的语义保持一致 https://ant.design/components/table-cn#components-table-demo-row-selection-custom

File: components/transfer/list.tsx (hunk 1 above)
Comment (li-jia-nan) [reply components/transfer/list.tsx]: done

File: components/transfer/index.tsx (hunk 2 above)
Comment (li-jia-nan) [reply components/transfer/index.tsx]: done

**Review by kiner-tang — APPROVED** [review]

**Review by MadCcc — APPROVED** [review]

**Pull request merged**